test/truefi2/lines-of-credit/TrueRateAdjuster.test.ts

```diff
@@ -1,33 +1,75 @@
 import { expect, use } from 'chai'
 import { Wallet } from 'ethers'
 
-import { beforeEachWithFixture, DAY, setupTruefi2 } from 'utils'
+import { beforeEachWithFixture, createApprovedLoan, DAY, parseEth, setupTruefi2 } from 'utils'
 
 import {
   TrueRateAdjuster,
   TrueFiPool2,
   TrueFiPool2__factory,
   TimeAveragedBaseRateOracle,
   TimeAveragedBaseRateOracle__factory,
+  TrueRatingAgencyV2,
+  MockTrueCurrency,
+  StkTruToken,
+  LoanFactory2,
+  TrueLender2,
+  TrueCreditAgency,
 } from 'contracts'
 
-import { solidity } from 'ethereum-waffle'
+import { MockProvider, solidity } from 'ethereum-waffle'
 
 use(solidity)
 
 describe('TrueRateAdjuster', () => {
   let owner: Wallet
   let borrower: Wallet
+  let provider: MockProvider
   let rateAdjuster: TrueRateAdjuster
+  let ratingAgency: TrueRatingAgencyV2
+  let tru: MockTrueCurrency
+  let stkTru: StkTruToken
+  let loanFactory: LoanFactory2
+  let tusdPool: TrueFiPool2
+  let tusd: MockTrueCurrency
+  let lender: TrueLender2
+  let creditAgency: TrueCreditAgency
 
   beforeEachWithFixture(async (wallets, _provider) => {
     [owner, borrower] = wallets
+    provider = _provider
 
     ;({
       rateAdjuster,
-    } = await setupTruefi2(owner, _provider))
+      rater: ratingAgency,
+      tru,
+      stkTru,
+      loanFactory,
+      standardToken: tusd,
+      standardPool: tusdPool,
+      lender,
+      creditAgency,
+    } = await setupTruefi2(owner, provider))
```

Comment:
Maybe we could use a mock TrueFiPool2 where we can manually set proFormaLiquidRatio. I *think* that's the only external state that TRA ever requires.

Reply:
There are more tests to come to this suite, that will be using it. I know it may be a little bit messy, but I always try to be as close to the real model as possible. Mocking has already caused some malfunctioning in the past, so I try to avoid it unless it's impossible to go on without it.

Reply:
Hmm. You make a good point regarding mocks. Sounds good, let's continue like this with setupTruefi2() and createApprovedLoan() then.